presto-orc/src/main/java/com/facebook/presto/orc/zstd/Huffman.java

```diff
@@ -0,0 +1,292 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.orc.zstd;
+
+import java.util.Arrays;
+
+import static com.facebook.presto.orc.zstd.BitStream.peekBitsFast;
+import static com.facebook.presto.orc.zstd.UnsafeUtil.UNSAFE;
+import static io.airlift.slice.SizeOf.SIZE_OF_INT;
+import static io.airlift.slice.SizeOf.SIZE_OF_SHORT;
+
+class Huffman
+{
+    private static final int MAX_SYMBOL = 255;
+    private static final int MAX_TABLE_LOG = 12;
+
+    // stats
+    private final byte[] weights = new byte[MAX_SYMBOL + 1];
+    private final int[] ranks = new int[MAX_TABLE_LOG + 1];
+
+    // table
+    private int tableLog;
+    private final byte[] symbols = new byte[1 << MAX_TABLE_LOG];
+    private final byte[] numbersOfBits = new byte[1 << MAX_TABLE_LOG];
+
+    private final FiniteStateEntropy decompressor = new FiniteStateEntropy(6);
```

Comment:
maybe name this `fse` so the call below becomes `fse.decompress`, or maybe full name `finiteStateEntropy`